runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useProcessActionVariables.ts

```diff
@@ -183,16 +183,21 @@ const useProcessActionVariables = (selectedElementParentId?: string) => {
             .filter((item: ActionVariableObject[]) => item.length > 0)
             .flat();
 
-        if(!selectedElementParentId) return { inputActionVariables: inputActionVars, outputActionVariables: outputActionVars };
+        const loopIds = allVarsAssigningActions
+            .filter((element: ScopedModdleElement) => element.actionId === 'loop.loop')
+            .map((element: ScopedModdleElement) => element.id);
+
+        const loopIterator = loopIds.includes(selectedElementParentId) ? [{ name: 'iterator', value: 'variable' }] : [];
+
+        if(!selectedElementParentId) return { inputActionVariables: inputActionVars, outputActionVariables: outputActionVars, loopVariables: [] };
```

Comment:
I think you can move 

if(!selectedElementParentId) return { inputActionVariables: inputActionVars, outputActionVariables: outputActionVars, loopVariables: [] };

over 

const loopIds = allVarsAssigningActions.......

because if selectedElementParentId does not exist you don't return neither loopIds nor loopIterator so unnecessary calculations take place